bioptim/interfaces/SolverOptions.py

```diff
@@ -0,0 +1,95 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+
+@dataclass
+class SolverOptions(ABC):
+    @abstractmethod
+    def set_convergence_tol(self, tol):
+        """
+         This function set the convergence tolerance
+        """
+
+    @abstractmethod
+    def set_constraint_tol(self, tol):
+        """
+         This function set the constraint tolerance.
+        """
+
+    @abstractmethod
+    def set_solver(self, name):
+        """
+         This function set the solver
+        """
+
+    def set_max_iter(self, num):
+        """
+         This function set the maximal number of iterations
+        """
+
+
+@dataclass
+class SolverOptionsIPOPT(SolverOptions):
+    tol: float = 1e-6  # default in ipopt 1e-8
+    dual_inf_tol: float = 1.0  # default in ipotpt 1
+    constr_viol_tol: float = 0.0001  # default in ipotpt 0.0001
+    compl_inf_tol: float = 0.0001  # default in ipotpt 0.0001
+    acceptable_tol: float = 1e-2  # default in ipopt 1e-2
+    acceptable_dual_inf_tol: float = 1e-2  # default in ipopt 1e-2
+    acceptable_constr_viol_tol: float = 1e-2  # default in ipopt 1e-2
+    acceptable_compl_inf_tol: float = 1e-2  # default in ipopt 1e-2
+    max_iter: int = 1000
+    hessian_approximation: str = "exact"  # "exact", "limited-memory"
+    limited_memory_max_history: int = 50
+    linear_solver: str = "mumps"  # "ma57", "ma86", "mumps"
+
+    def set_convergence_tol(self, val):
+        self.tol = val
+        self.compl_inf_tol = val
+        self.acceptable_tol = val
+        self.acceptable_compl_inf_tol = val
+
+    def set_constraint_tol(self, val):
+        self.constr_viol_tol = val
+        self.acceptable_constr_viol_tol = val
+
+    def set_solver(self, name):
+        self.linear_solver = name
```

Comment:
The way the function are created, makes this function useless (if we remove the supermethod), therefore I suggest to simply remove this